# EDA for ICU_note_dataset

#### Import dataset

In [2]:
import pandas as pd

df = pd.read_csv('dataset/ICU_note_dataset.csv')
df

/tmp/ipykernel_81804/991307044.py:3: DtypeWarning: Columns (4,5) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv('dataset/ICU_note_dataset.csv')


ROW_ID  SUBJECT_ID   HADM_ID   CHARTDATE            CHARTTIME  \
0            174       22532  167853.0  2151-08-04                  NaN   
1            175       13702  107527.0  2118-06-14                  NaN   
2            176       13702  167118.0  2119-05-25                  NaN   
3            177       13702  196489.0  2124-08-18                  NaN   
4            178       26880  135453.0  2162-03-25                  NaN   
...          ...         ...       ...         ...                  ...   
2083175  2070657       31097  115637.0  2132-01-21  2132-01-21 03:27:00   
2083176  2070658       31097  115637.0  2132-01-21  2132-01-21 09:50:00   
2083177  2070659       31097  115637.0  2132-01-21  2132-01-21 16:42:00   
2083178  2070660       31097  115637.0  2132-01-21  2132-01-21 18:05:00   
2083179  2070661       31097  115637.0  2132-01-21  2132-01-21 18:05:00   

                   STORETIME           CATEGORY DESCRIPTION     CGID  ISERROR  \
0                        NaN  Discharge summary      Report      NaN      NaN   
1                        NaN  Discharge summary      Report      NaN      NaN   
2                        NaN  Discharge summary      Report      NaN      NaN   
3                        NaN  Discharge summary      Report      NaN      NaN   
4                        NaN  Discharge summary      Report      NaN      NaN   
...                      ...                ...         ...      ...      ...   
2083175  2132-01-21 03:38:00      Nursing/other      Report  17581.0      NaN   
2083176  2132-01-21 09:53:00      Nursing/other      Report  19211.0      NaN   
2083177  2132-01-21 16:44:00      Nursing/other      Report  20104.0      NaN   
2083178  2132-01-21 18:16:00      Nursing/other      Report  16023.0      NaN   
2083179  2132-01-21 18:31:00      Nursing/other      Report  16023.0      NaN   

                                                      TEXT  
0        Admission Date:  [**2151-7-16**]       Dischar...  
1        Admission Date:  [**2118-6-2**]       Discharg...  
2        Admission Date:  [**2119-5-4**]              D...  
3        Admission Date:  [**2124-7-21**]              ...  
4        Admission Date:  [**2162-3-3**]              D...  
...                                                    ...  
2083175  NPN\n\n\n#1  Infant remains in RA with O2 sats...  
2083176  Neonatology\nDOL #5, CGA 36 weeks.\n\nCVR: Con...  
2083177  Family Meeting Note\nFamily meeting held with ...  
2083178  NPN 1800\n\n\n#1 Resp: [**Known lastname 2243*...  
2083179  NPN 1800\nNursing Addendum:\n[**Known lastname...  

[2083180 rows x 11 columns]

#### Preprocess dataset

The dataset has 2083180 entries. First, narrow the dataset down to just the TEXT column, which contains the freeform text, and further to doctor's notes (where the column CATEGORY="Physician"). 

In [3]:
filtered = df[df['CATEGORY'] == "Physician "]['TEXT']

#### Extract subset to explore

It is still too large to view with a conventional csv viewer, so one must reduce the size of the dataset further. One option is to filter further with the `pandas` library for certain keywords. Another option is to simply print a small selection of the dataset to a file. This is accomplished by defining `s` and `e` to be the start and end indices of the subset. Scroll through `icu_record.txt` to find data samples with appropriate data.

In [4]:
s=0
e=100

with open('icu_record_subset.txt', 'w') as f:
    for i in range(s, e):
        f.write(f"Record {s+i}:\n")
        f.write(filtered.iloc[i])
        f.write('\n################################################################\n\n')


#### Select Samples for Analysis

Put the indices for the selected samples in the array `r`, and save each one in a new .txt file in a folder for the desired metric. Now, you can use `run_model.py` on them with a specified metric.

In [7]:
import os
r = [49, 50, 2, 54]

for i in range(len(r)):
    fn = f'qSOFA_records/qSOFA{i+1}.txt'
    if os.path.isfile(fn): continue
    with open(fn, 'w') as f:
        f.write(filtered.iloc[r[i]])